panda/crypto/panda_test.go

```diff
@@ -209,23 +211,26 @@ func NoTestUpdateSRV(t *testing.T) {
 func runKXWithSerialize(resultChan chan interface{}, log *logging.Logger, mp MeetingPlace, srv []byte, secret []byte, message []byte) {
 
 	shutdownChan := make(chan struct{})
-	go func() {
-		<-shutdownChan
-	}()
 	contactID := uint64(123)
 	pandaChan := make(chan PandaUpdate)
 	kx, err := NewKeyExchange(rand.Reader, log, mp, srv, secret, message, contactID, pandaChan, shutdownChan)
 
+	if err != nil {
+		resultChan <- err
+		return
+	}
+	kx.log = log
+	go kx.Run()
+
 	go func() {
 		var reply []byte
 		for {
 			pandaUpdate := <-pandaChan
 			close(shutdownChan)
+			shutdownChan = make(chan struct{})
```

Comment:
should we maybe just use a different name for this chan? it seems a bit confusing that we overwrite the old one?